parquet-hadoop/src/main/java/parquet/hadoop/ParquetOutputFormat.java

```diff
@@ -299,6 +312,7 @@ private CompressionCodecName getCodec(TaskAttemptContext taskAttemptContext) {
       LOG.warn("The configuration " + MEMORY_POOL_RATIO + " has been set. It should not " +
           "be reset by the new value: " + maxLoad);
     }
+    memoryManager.registerCallBack(callBack);
```

Comment:
I think it makes sense to change Hive, but Parquet doesn't really need to count the number of scale-down events.